modules/ROOT/pages/6.4.1-release-notes.adoc

```diff
@@ -0,0 +1,1117 @@
+//#TINY-7993
+
+A bug was found within the context menu and dialog when inserting `links` in previous versions of {productname}. The issue resulted in the `open link` being deactivated within the context menu and dialog when the user would execute a specific chain of clicks. Such as:
+
+.Example
+====
+. Opening the editor
+. Type “test1 test2 test3” into the editor
+. Double click to select “test2”
+. Use the context menu to insert a new link over the text “test2”
+. Add a valid URL
+. Press `Enter`
+. Without taking any other action, right click on “test2”
+
+In this example the bug would deactivate the `open link` button within the context menu and dialog, which removes the users ability to reopen the “open link” dialog.
+====
+
+The fix now allows all three link-plugins (Link, Remove link and Open link) to work within the context menu items.
+
+In {productname} 6.4.1, when the user executes right `mouse click` on the link selection immediately after attaching a link to a `<string>`, the link-plugins now work as expected.
+
+=== Editor commands ForwardDelete and Delete were deleting contents inside noneditable elements
+//#TINY-9477
+
+Previously, when using the `ForwardDelete` or `Delete` commands in earlier versions of {productname}, the contents within `contenteditable="false"` elements were deleted. This meant that if there were nested elements within a `contenteditable="false"` root element that were also set to `contenteditable="false"`, they could be deleted from the root element.
+
+To resolve this issue, {productname} 6.4.1 included several enhancements to prevent all commands and keyboard functions, including delete/forwarddelete, from affecting `contenteditable="false"` content.
+
+Consequently, in {productname} 6.4.1, using the forwarddelete or delete keys on `contenteditable="false"` elements within a `contenteditable="false"` element will no longer delete the contents within the `contenteditable="false"` elements.
+
+=== Backspace or delete keys was deleting contents inside noneditable elements
+//#TINY-9477
+
+In the previous versions of {productname}, when utilizing the backspace or delete keys, they would delete the contents inside `contenteditable="false"` elements.
+
+As a consequence, elements nested within a `contenteditable="false"` root element set to `contenteditable="false"`, were able to be removed/deleted from the `contenteditable="false"` root element. To fix this issue, {productname} 6.4.1 introduced various improvements to block all commands and keyboard functions such as `delete/backspace` from being applied to `contenteditable="false"` content. 
+
+
+As a result, in {productname} 6.4.1, when using the backspace or delete function keys on `contenteditable="false"` elements within a `contenteditable="false"` element, the contents within the `contenteditable="false"` elements will no longer be deleted. 
+
```

Comment:
```suggestion
=== Editor commands ForwardDelete, Backspace and Delete were deleting contents inside`contenteditable="false"` elements
//#TINY-9477

Previously, when using the ForwardDelete, Backspace or Delete commands in earlier versions of {productname}, the contents of a `contenteditable="false"` nested element would be deleted.

As a consequence, if a `contenteditable="false"` element was nested within a `contenteditable="false"` root element, the contents of the nested element would be deleted from the root element when the specified commands were executed.

To resolve this issue, {productname} 6.4.1 included several enhancements to prevent all commands and keyboard functions which include ForwardDelete, Backspace and Delete commands from affecting `contenteditable="false"` content.

```

Reply:
@TheSpyder Please see above suggested merge for the above two sections.

Reply:
Almost. `backspace` and `delete` are keys on the keyboard, `Delete` and `ForwardDelete` are commands in the editor. You've made Backspace sound like a command and incorrectly combined the two delete concepts.

Reply:
@TheSpyder Please see above edited changes.

Reply:
Minor corrections and fixing a typo

```
=== Content could be deleted inside `contenteditable="false"` elements
//#TINY-9477
When content was deleted (either through the ForwardDelete/Delete commands, or the user pressing Backspace/Delete keys) in earlier versions of {productname}, the contents of a `contenteditable="false"` nested element would be deleted.
As a consequence, if a `contenteditable="false"` element was nested within a `contenteditable="false"` root element, the contents of the nested element would be deleted from the root element when the specified command or keys were executed.
To resolve this issue, {productname} 6.4.1 includes several enhancements to prevent all commands and keyboard functions from affecting `contenteditable="false"` content which includes the ForwardDelete/Delete commands and both the Backspace and Delete keys.
```